pandas/core/arrays/boolean.py

```diff
@@ -773,19 +773,20 @@ def kleene_or(
 
     raise_for_nan(right, method="or")
 
-    mask = left_mask
+    result = left | right
 
     if right_mask is not None:
-        mask = mask | right_mask
+        # output is unknown where (False & NA), (NA & False), (NA & NA)
+        mask = (
+            ((~left & ~left_mask) & right_mask)
```

Comment:
the "left False" can also be done as `~(left | left_mask)`, which is a little bit faster (one operation less) 

(and same for "right false" below)